R/zchunk_LA100.IEA_downscale_ctry.R

```diff
@@ -0,0 +1,308 @@
+        left_join_error_no_match(L100.Composites_pop, by = c("IEAcomposite", "year")) %>%
+        mutate(pop_share = value / composite_population) %>%
+        # drop extraneous columns
+        select(iso, year, pop_share) ->
+        L100.Others_pop_share
+
+      # Spread L100.Others_pop_share to wide format in preparation for computing L100.Others_ctry_bal
+      # This is relatively inexpensive
+      L100.Others_pop_share <- spread(L100.Others_pop_share, year, pop_share)
+
+      # Multiply the repeated country databases by the population shares to get the energy balances by country (159-162)
+      L100.Others_repCtry %>%
+        select(iso, FLOW, PRODUCT, COUNTRY) %>%
+        left_join_error_no_match(L100.Others_repCtry, by = c("iso", "FLOW", "PRODUCT", "COUNTRY")) %>%
+        rename(IEAcomposite = COUNTRY) ->
+        L100.Others_ctry_bal
+
+      L100.Others_repCtry %>%
+        select(iso) %>%
+        left_join_error_no_match(L100.Others_pop_share, by = "iso") %>%
+        select(-iso) ->
+        pop_share
+
+      hyc <- which(names(L100.Others_ctry_bal) %in% HISTORICAL_YEARS)
+      L100.Others_ctry_bal[hyc] <- L100.Others_ctry_bal[hyc] * pop_share
+
+      # Subset each of these final energy balances to only the rows that aren't zero in all years
+      L100.Others_ctry_bal[rowSums(L100.Others_ctry_bal[hyc]) !=0, ] %>%
+        select(-IEAcomposite) ->
+        L100.Others_ctry_bal
+
+      hyc <- which(names(L100.IEAsingle) %in% HISTORICAL_YEARS)
+      L100.IEAsingle[rowSums(L100.IEAsingle[hyc]) != 0, ] %>%
+        select(-COUNTRY) %>%
+        filter(! iso %in% L100.USSR_Yug_ctry_bal$iso) ->
+        L100.IEAsingle_noUSSR_Yug
+
+      hyc <- which(names(L100.USSR_Yug_ctry_bal) %in% HISTORICAL_YEARS)
+      L100.USSR_Yug_ctry_bal[rowSums(L100.USSR_Yug_ctry_bal[hyc]) !=0, ] %>%
+        select(-IEAcomposite) ->
+        L100.USSR_Yug_ctry_bal
+
+      # Combine the country-level data tables and write out energy balances (using iso codes rather than IEA's country names)
+      bind_rows(L100.IEAsingle_noUSSR_Yug,
+                # TODO: we end up with two rows for each iso, FLOW, PRODUCT in the following data frame
+                L100.USSR_Yug_ctry_bal,
+                L100.Others_ctry_bal) %>%
+        add_title("IEA energy balances downscaled to 201 countries by iso code, FLOW, PRODUCT, and historical year") %>%
+        add_units("ktoe and GWh") %>%
+        add_comments("Combine OECD and non-OECD data; perform upfront adjustments for other Africa, Turkey, and South Africa;") %>%
+        add_comments("split out and handle the 1990 split of Yugoslavia and USSR; use population to downscale IEA composite regions") %>%
+        add_comments("to individual countries; filter out countries without data in any year.") ->
+        L100.IEA_en_bal_ctry_hist
+
+    } else {
+      # raw IEA datasets not available, so used presaved data loaded above
+      if(isTRUE(is.na(L100.IEA_en_bal_ctry_hist_preprocessed))) {
```

Comment:
No! At least as per existing agreements. `L100.IEA_en_bal_ctry_hist` is basically the same as committing the IEA energy balances; it's the same dataset, expanded and made more usable. That said, it is five years out of date, so they might allow us to distribute it? Sometimes they make old datasets freely available. But there is no practical difference, for us or for them, between the output of this chunk and its inputs (en_OECD, en_nonOECD).